### ISSUE TITLE
Inference 단계에서 반복 토큰 생성(mode collapse) 발생

### ISSUE THREAD

**Issue by semin-go** [semin-go/NLP#4]
학습이 완료된 모델을 이용해 추론(test.py)을 수행할 때,
의미 없는 토큰(예: 여러분이, 갑자기, 原原)이 반복 생성되는 문제가 발생함.

이는 학습 과정 중 gradient explosion 및 불안정한 디코딩 설정으로 인해
모델이 특정 패턴에 수렴하는 mode collapse 현상으로 판단됨.

원인
-작은 T5 모델에서 상대적으로 큰 learning rate 사용
-학습 초반 gradient 폭주
-eos 토큰 이전에 반복 패턴 생성

**Comment by semin-go** [comment]
해결 방안
learning rate 감소
gradient clipping 강화
generation 시 no_repeat_ngram_size, repetition_penalty 적용

**Comment by semin-go** [comment]
<img width="1778" height="278" alt="Image" src="https://github.com/user-attachments/assets/65769a73-fbd2-44bd-bf6f-dcc7ea08ab94" />
